python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -1508,3 +1551,28 @@ def info_string(self):
         autoscaler_summary = self.summary()
         assert autoscaler_summary
         return "\n" + format_info_string(lm_summary, autoscaler_summary)
+
+    def get_one_worker_node_config_obj_ref(self):
+        """
+        When DRA is enabled, only one worker node type is enough, so in this function
+        we just pick the first worker node showed up in the dict.
+        """
+        head_node_type = self.config.get("head_node_type")
+        whatever_worker_node_config = None
+        for node_type in self.all_node_types:
+            if node_type == head_node_type:
+                continue
+            whatever_worker_node_config = self.available_node_types[node_type]
+            break
+        return whatever_worker_node_config
+
+    def dra_triggered_recently(self):
+        now = time.time()
+        if not self.last_resource_adjust_time:
+            self.last_resource_adjust_time = now
+            return True
+        # the unit of time.time() is second
+        delta = now - self.last_resource_adjust_time
+        if delta > (60 * self.config["idle_timeout_minutes"]):
```

Comment:
I think this existing value is pretty feasiable.
This value defines the idle time for the cluster to wait before it starts to shrink
When the cluster starts to shrink, it means it has been idel for a period. Then it also can be the time to reset the dynamically enlarged CPU/Memory. After all the cluster has been idel for a while, it's time to do the reset the reources being occuppied by the min number of workers. 
Also this can reduce another definition for a variable for vSphere only.